### DISCUSSION TITLE
[BUG] ADDON variables in .env not working

### DISCUSSION THREAD

**Discussion by micadix** [jonaswinkler/paperless-ng#632] [Support]
i installed paperless-ng on a fresh ubuntu server instance with "install-paperless-ng.sh" works all fine. but i tried to give paperless 2 environment variables:
i added the following to "docker-compose.env":
PAPERLESS_FILENAME_FORMAT={created_year}/{created}-{correspondent}-{title}
PAPERLESS_TIME_ZONE=Europe/Berlin

but my file names are generated like: e.g. "0000005.pdf"

any hints for me? also timezone seems not to work. looks like it is ignoring my additional 2 lines in the env-file...

**Comment by jonaswinkler** [comment]
Did you restart paperless-ng after making these changes?

**Comment by micadix** [comment]
i rebooted the complete vm so i think this restarts als the docker right?

**Comment by jonaswinkler** [comment]
Probably. `PAPERLESS_FILENAME_FORMAT` looks good to me. Does saving any existing document changes the filename properly?

Regarding the time zone: What makes you think that it's not working properly?

**Comment by micadix** [comment]
> Does saving any existing document changes the filename properly?

what do u mean?

if i look into tasks in the filed "next ruN" it says "you are 1 hour ahead the server time"...

**Comment by micadix** [comment]
paperless is just ignoring my inputs in the .env file...is there another way to get this working?

**Comment by jonaswinkler** [comment]
The other place to make these changes would be the .yml file itself. Look for the section that defines `PAPERLESS_REDIS` (and possibly more depending on what options you've chosen). You can make changes there.

**Comment by micadix** [comment]
okay i made the changes here, rebooted - nothing happend...

my files are in /home/username/paperless-ng/docker-compose.yml

i edited here... :(

**Comment by jonaswinkler** [comment]
Not exactly sure what's going on. Please try to explicitly restart paperless with `docker-compose down` and `docker-compose up -d`.

**Comment by rknightion** [comment]
FYI @micadix as it's a common misconception with Docker; just restarting a container won't pick up changes to environment variables. Neither will restarting the vm.
You need to completely remove and recreate the container, as Jonas said above with a docker-compose down and up (in recent versions of compose just running an up -d will recreate any container with changes as well).

From https://docs.docker.com/compose/reference/restart/

> If you make changes to your docker-compose.yml configuration these changes are not reflected after running this command. For example, changes to environment variables (which are added after a container is built, but before the container’s command is executed) are not updated after restarting.
